File: backend/src/models/Doctor.js
const { Model, DataTypes } = require('sequelize');

class Doctor extends Model {
  static initModel(sequelize) {
    Doctor.init({

    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false
    },
    email: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true,
      validate: {
        isEmail: true
      }
    },
    password: {
      type: DataTypes.STRING,
      allowNull: false
    },
    phone: {
      type: DataTypes.STRING,
      allowNull: false
    },
    specializations: {
      type: DataTypes.ARRAY(DataTypes.STRING),
      allowNull: false,
      defaultValue: []
    },
    qualifications: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    experience: {
      type: DataTypes.INTEGER,
      allowNull: false,
      validate: {
        min: 0
      }
    },
    bio: {
      type: DataTypes.TEXT,
      allowNull: true
    },
    gender: {
      type: DataTypes.ENUM('male', 'female', 'other'),
      allowNull: false
    },
    dateOfBirth: {
      type: DataTypes.DATEONLY,
      allowNull: false
    },
    address: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    city: {
      type: DataTypes.STRING,
      allowNull: false
    },
    state: {
      type: DataTypes.STRING,
      allowNull: false
    },
    country: {
      type: DataTypes.STRING,
      allowNull: false
    },
    pincode: {
      type: DataTypes.STRING,
      allowNull: false
    },
    profileImage: {
      type: DataTypes.STRING,
      allowNull: true
    },
    licenseNumber: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true
    },
    hospital: {
      type: DataTypes.STRING,
      allowNull: true
    },
    consultationFee: {
      type: DataTypes.DECIMAL(10, 2),
      allowNull: true,
      defaultValue: 0.00
    },
    isAvailable: {
      type: DataTypes.BOOLEAN,
      defaultValue: true
    },
    rating: {
      type: DataTypes.DECIMAL(3, 2),
      defaultValue: 0.00,
      validate: {
        min: 0,
        max: 5
      }
    },
    totalRatings: {
      type: DataTypes.INTEGER,
      defaultValue: 0
    }
  
    }, {
      sequelize,
      modelName: 'Doctor',

    timestamps: true
  
    });
    return Doctor;
  }

  static associate(models) {
    // Associations will be defined in models/index.js
  }
}

module.exports = Doctor;

File: fastapi_backend/models/doctor.py
from beanie import Document
from pydantic import BaseModel, EmailStr, Field, validator
from typing import Optional, List
from datetime import datetime, date
from enum import Enum
from decimal import Decimal

class Gender(str, Enum):
    MALE = "male"
    FEMALE = "female"
    OTHER = "other"

# Pydantic schemas
class DoctorBase(BaseModel):
    name: str
    email: EmailStr
    phone: str
    specializations: List[str] = []
    qualifications: str
    experience: int = Field(..., ge=0)
    bio: Optional[str] = None
    gender: Gender
    date_of_birth: date
    address: str
    city: str
    state: str
    country: str
    pincode: str
    profile_image: Optional[str] = None
    license_number: str
    hospital: Optional[str] = None
    consultation_fee: Optional[Decimal] = Field(default=Decimal("0.00"), ge=0)
    is_available: bool = True
    rating: Decimal = Field(default=Decimal("0.00"), ge=0, le=5)
    total_ratings: int = 0

class DoctorCreate(DoctorBase):
    password: str

class DoctorUpdate(BaseModel):
    name: Optional[str] = None
    phone: Optional[str] = None
    specializations: Optional[List[str]] = None
    qualifications: Optional[str] = None
    experience: Optional[int] = Field(None, ge=0)
    bio: Optional[str] = None
    gender: Optional[Gender] = None
    date_of_birth: Optional[date] = None
    address: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    country: Optional[str] = None
    pincode: Optional[str] = None
    profile_image: Optional[str] = None
    license_number: Optional[str] = None
    hospital: Optional[str] = None
    consultation_fee: Optional[Decimal] = Field(None, ge=0)
    is_available: Optional[bool] = None
    rating: Optional[Decimal] = Field(None, ge=0, le=5)
    total_ratings: Optional[int] = None

class DoctorResponse(DoctorBase):
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class DoctorLogin(BaseModel):
    email: EmailStr
    password: str

# Beanie Document
class Doctor(Document):
    name: str
    email: EmailStr
    password: str
    phone: str
    specializations: List[str] = []
    qualifications: str
    experience: int = Field(..., ge=0)
    bio: Optional[str] = None
    gender: str
    date_of_birth: str  # Store as string for now
    address: str
    city: str
    state: str
    country: str
    pincode: str
    profile_image: Optional[str] = None
    license_number: str
    hospital: Optional[str] = None
    consultation_fee: float = Field(default=0.0, ge=0)
    is_available: bool = True
    rating: float = Field(default=0.0, ge=0, le=5)
    total_ratings: int = 0
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "doctors"
        indexes = [
            "email",
            "license_number",
            "specializations",
            "city",
            "state",
            "is_available",
            "rating"
        ]

    def __str__(self):
        return f"Doctor(id={self.id}, email={self.email}, name={self.name})"
